src/lava/lib/optimization/apps/scheduler/problems.py

```diff
@@ -0,0 +1,310 @@
+    def num_tasks(self):
+        return self._num_tasks
+
+    @num_tasks.setter
+    def num_tasks(self, val: int):
+        self._num_tasks = val
+
+    @property
+    def task_ids(self):
+        return self._task_ids
+
+    @property
+    def task_attrs(self):
+        return self._task_attrs
+
+    @task_attrs.setter
+    def task_attrs(self, attr_vec):
+        self._task_attrs = attr_vec
+
+    @property
+    def sat_cutoff(self):
+        return self._sat_cutoff
+
+    @sat_cutoff.setter
+    def sat_cutoff(self, val: float):
+        self._sat_cutoff = val
+
+    @property
+    def random_seed(self):
+        return self._random_seed
+
+    @random_seed.setter
+    def random_seed(self, val: int):
+        self._random_seed = val
+
+    def is_node_valid(self, *args):
+        return True
+
+    def is_edge_conflicting(self, node1, node2):
+        nodes = self.graph.nodes
+        is_same_agent = (nodes[node1]["agent_id"] == nodes[node2]["agent_id"])
+        is_same_task = (nodes[node1]["task_id"] == nodes[node2]["task_id"])
+        return True if is_same_agent or is_same_task else False
+
+    def generate(self, seed=None):
+        """ Generate a new scheduler problem. """
+        if seed != self.random_seed:  # set seed only if it's different
+            self.random_seed = seed
+            np.random.seed(seed)
+        self.graph = ntx.Graph()
+        self.generate_valid_nodes()
+        self.generate_edges_from_constraints()
+        self.rescale_adjacency()
+
+    def generate_valid_nodes(self):
+        node_id = 0
+        if self.agent_attrs is None:
+            self.agent_attrs = np.reshape(self.agent_ids,
+                                          (len(self.agent_ids), 1))
+        agent_id_attr_map = dict(zip(self.agent_ids, self.agent_attrs))
+        if self.task_attrs is None:
+            self.task_attrs = (
+                np.tile(np.reshape(self.task_ids,
+                                   (len(self.task_ids), 1)), (1, 2)))
+        task_id_attr_map = dict(zip(self.task_ids, self.task_attrs))
+        for aid, a_attr in agent_id_attr_map.items():
+            for tid, t_attr in task_id_attr_map.items():
+                if self.is_node_valid(aid, tid):
+                    self.graph.add_node(node_id,
+                                        agent_id=aid,
+                                        task_id=tid,
+                                        agent_attr=a_attr,
+                                        task_attr=t_attr)
+                    node_id += 1
+
+    def generate_edges_from_constraints(self):
+        num_nodes = len(self.graph.nodes)
+        self.adjacency = (
+            np.zeros((num_nodes, num_nodes), dtype=int))
+        for n1 in self.graph.nodes:
+            for n2 in self.graph.nodes:
+                not_same = n1 != n2
+                is_conflict = self.is_edge_conflicting(n1, n2)
+                if not_same and is_conflict:
+                    self.graph.add_edge(n1, n2)
```

Comment:
Is there any way to prevent that people accidentally call generate_edges_from_constraints twice, and the graph will have a full set of redundant edges?

Reply:
If you want to add multiple edges, it might be more efficient to pre-calculate and buffer all node tuples first, and then call add_edges_from to add all edges simultaneously.

Reply:
Made this method private (see an earlier comment). This is solely called inside the `generate` method. So, ideally, it should not be used outside of this class.